StRoot/StFwdTrackMaker/StFwdTrackMaker.cxx

```diff
@@ -258,7 +258,7 @@ int StFwdTrackMaker::Finish() {
 
     mForwardTracker->finish();
 
-    gDirectory = prevDir;
+    prevDir->cd();
```

Comment:
Agreed: we shouldn't be setting global ROOT variables ourselves.

Reply:
Do you see it as a problem with ROOT6 only?

Reply:
gDirectory = prevDir compiles under root 5, but fails to compile under root 6.24.  I looked into the root 6.24 code.  They deleted the assignment operator, and made other changes to make TDirectory thread safe.